elftools/elf/elffile.py

```diff
@@ -120,7 +120,10 @@ def iter_sections(self):
     def num_segments(self):
         """ Number of segments in the file
         """
-        return self['e_phnum']
+        if self['e_phnum'] < 0xffff:
```

Comment:
Could you please add a comment here with the relevant text of the spec and a link to the spec section?